letslambda.json

```diff
@@ -6,46 +6,133 @@
             "Description": "S3 Bucket name (not arn)",
             "Type": "String"
         },
-	"Region": {
+        "Region": {
             "Description": "Region short code name where the S3 bucket is located (ie: eu-west-1)",
             "Type": "String"
-	}
+        },
+        "KmsEncryptionKeyArn": {
+            "Description": "Default KMS Encryption Key (arn) used to securely store your SSL private keys. Use 'AES256' for S3 automatic encryption",
+            "Type": "String"
+        }
+    },
+    "Conditions": {
+        "KmsArnProvided": {"Fn::Not": [{"Fn::Equals": [{"Ref": "KmsEncryptionKeyArn"}, "AES256"]}]}
     },
     "Resources" : {
-        "LetsLambdaManagedPolicy": {
+        "DenyUnEncryptedPrivKeyUploadsBucketPolicy" : {
+            "Type" : "AWS::S3::BucketPolicy",
+            "Properties" : {
+                "Bucket" : {"Ref" : "Bucket"},
+                "PolicyDocument": {
+                    "Statement":[{
+                        "Sid": "DenyUnEncryptedPrivKeyUploads",
+                        "Action": ["s3:PutObject"],
+                        "Effect": "Deny",
+                        "Resource": { "Fn::Join" : ["", ["arn:aws:s3:::", { "Ref" : "Bucket" } , "/*.key.pem" ]]},
```

Comment:
I think it would be a good idea to introduce a parameter `S3BasePath` so that the IAM could be more targetted - this arrangement will affect the whole bucket, which will limit the use of letslambda in accounts that don't want to devote a whole bucket to it (there's a hard upper limit of 100 buckets per account because the names are globally unique).